src/main/kotlin/org/opensearch/commons/alerting/model/Monitor.kt

```diff
@@ -55,7 +56,7 @@ data class Monitor(
 
             require(triggerIds.add(trigger.id)) { "Duplicate trigger id: ${trigger.id}. Trigger ids must be unique." }
             // Verify Trigger type based on Monitor type
-            when (monitorType) {
+            when (MonitorType.valueOf(monitorType.uppercase(Locale.ROOT))) {
```

Comment:
try-catch. remote monitor's type will throw Exception right?

Reply:
why not convert enum to strings and do contains check on list of enum strings

Reply:
fixed this.